src/Accelerators/CMakeLists.txt

```diff
@@ -1,43 +1,47 @@
 # SPDX-License-Identifier: Apache-2.0
+
 set(ACCEL_LIST "")
-file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/AcceleratorKinds.hpp "// Enumeration of accelerators \n")
+
+# Generate the enumerators used to identify the kind of accelerators.
+set(AcceleratorKinds "${CMAKE_CURRENT_BINARY_DIR}/AcceleratorKinds.inc")
+file(WRITE "${AcceleratorKinds}" "// Enumeration of accelerators.\n")
```

Comment:
As I also wrote in an other PR, I find this code gen in the CMake hard to read. I suggest to simply generate a file of macros, like we do for `OM_TYPE_METADATA_DEF`. We would build here a single file with all of the needed info in a single macro, and then generate the right code using a #define . I understand it is a matter of preference.

Reply:
@AlexandreEichenberger  not sure I follow. Currently we have a typed enum in the Accelerator class that includes a file generated by this cmake file:
```
class Accelerator {
public:
  /// Kinds of accelerators.
  // clang-format off
  enum class Kind {
  // Note: AcceleratorKinds.inc is generated by cmake.
  #include "src/Accelerators/AcceleratorKinds.inc"
  };
  // clang-format on
```
We had discussions with @caoimhinuibrian and settled on generating the enumerators for each accelerators because we did not want to expose the accelerators details directly in the Accelerator header file. 

The key design point was that we wanted a cmake based solution because we do not know how many accelerators (if any) should be "compiled" in. With that design decision in mind, could you explain your proposed change?